### PR TITLE
VS1-vertical-slice

### PR DESCRIPTION
**Pull request by BIGduzy**
[VS1](https://trello.com/c/95m3Fa1v/16-vs1-vertical-slice) for sprint 3.

### CODE REVIEW

File: src/motor_interface.hpp [hunk 1]
```diff
@@ -21,25 +22,49 @@ class MotorInterface {
...
+    uint8_t speed = 0;
+
   public:
     /**
      * @brief Default constructor
      */
     MotorInterface();
+
     /**
      * @brief getter for selectedInterface
```
Comment (vasilishomegas) [inline src/motor_interface.hpp:44]: Check wiki for proper doxygen documentation, you need to document the return value. This goes for the doxygen of other functions as well.

File: src/motor_interface.hpp [hunk 2]
```diff
@@ -21,25 +22,49 @@ class MotorInterface {
...
+    void setDirection(const DIRECTION direction);
+    /**
+     * @brief getter for speed
+     */
+    uint8_t getSpeed() const;
+    /**
+     * @brief setter for speed
+     */
+    void setSpeed(const uint8_t speed);
 };
```
Comment (vasilishomegas) [inline src/motor_interface.hpp:67]: Check you indentation, run clang-format.

File: src/motor_interface.hpp [hunk 3]
```diff
@@ -21,25 +22,49 @@ class MotorInterface {
...
   private:
     ///< The interface that is currently selected
     INTERFACE selectedInterface = INTERFACE::DC;
 
+    ///< The direction of the motor rotation, forward by default
+    DIRECTION direction = DIRECTION::FORWARD;
+    ///< Motor rotation speed, 0 - 100%
+    uint8_t speed = 0;
+
   public:
```
Comment (vasilishomegas) [inline src/motor_interface.hpp:37]: I see public - private - public.

File: src/motor_interface.hpp [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
 #include "wrap-hwlib.hpp"
 
+namespace MotorController {
```
Comment (vasilishomegas) [inline src/motor_interface.hpp:13]: Indentation of everything within the namespace

**Review by vasilishomegas — CHANGES_REQUESTED** [review]

**Comment by vasilishomegas** [comment]
Travis is unhappy, complexity_test and clangformat_test fail.

File: src/motor_interface.hpp (hunk 4 above)
Comment (vasilishomegas) [reply src/motor_interface.hpp:13]: So apparently this is how it should be...

File: src/motor_interface.hpp (hunk 2 above)
Comment (vasilishomegas) [reply src/motor_interface.hpp:67]: So apparently this is how it should be...

**Comment by BIGduzy** [comment]
There will be a meeting with the leads about the doxygen for getters and setters on mondag, for now you can ignore them.

**Comment by BIGduzy** [comment]
No, not yet.

**Review by vasilishomegas — APPROVED** [review]

**Pull request merged**